clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1051,7 +1061,8 @@ private void close(long timeout, TimeUnit timeUnit, boolean swallowException) {
                     try {
                         this.ioThread.join(timeUnit.toMillis(timeout));
                     } catch (InterruptedException t) {
-                        firstException.compareAndSet(null, t);
+                        Thread.currentThread().interrupt();
```

Comment:
Since we interrupt in the constructor of `InterruptException`, we could drop this. Same below.

Reply:
Good point. Didn't know we interrupt in the constructor. Good to know :)